src/sentry/utils/relocation.py

```diff
@@ -308,9 +324,80 @@ class OrderedTask(Enum):
 )
 
 
-# A custom logger that roughly matches the parts of the `click.echo` interface that the
-# `import_*` methods rely on.
+class StorageBackedCheckpointExporter(ExportCheckpointer):
+    """
+    An export checkpointer that uses GCP cloud storage to store encrypted checkpoints for every
+    model we export for a SAAS_TO_SAAS relocation.
+    """
+
+    def __init__(
+        self,
+        *,
+        crypto: EncryptorDecryptorPair,
+        uuid: UUID,
+        storage: Storage,
+    ):
+        self.__crypto = crypto
+        self.__uuid = uuid
+        self.__storage = storage
+
+    def _get_path_name(self, model_name: NormalizedModelName) -> str:
+        return f"runs/{self.__uuid}/saas_to_saas_export/_checkpoints/{str(model_name)}.enc.tar"
+
+    def get(self, model_name: NormalizedModelName) -> RpcExportOk | None:
+        logger_data: dict[str, Any] = {"uuid": str(self.__uuid), "model": str(model_name)}
+        path_name = self._get_path_name(model_name)
+        try:
+            with self.__storage.open(path_name, "rb") as fp:
+                logger_data["encrypted_contents_size"] = fp.tell()
+                json_data = decrypt_encrypted_tarball(fp, self.__crypto.decryptor)
+                parsed_json = self._parse_cached_json(json_data)
+                if parsed_json is None:
+                    logger.info(
+                        "Export checkpointer: miss",
+                        extra=logger_data,
```

Comment:
Just wanted to sanity check: There will be no sensitive data exposed here?

Reply:
Nope: just the name of the model, the UUID of the relocation (safe), and the size (in bytes) of the encrypted data used in the checkpoint. None of that is sensitive.